TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs

```diff
@@ -51,7 +51,7 @@ public Lane(Road road, RoadNode roadNode, LaneType type, VertexPath path)
             {
                 // Update the prev and create a new current node
                 prev = curr;
-                curr = new LaneNode(path.GetPoint(i), GetNodeRotation(path.cumulativeLengthAtEachVertex[i], _road.EndOfPathInstruction), currRoadNode, prev, null);
+                curr = new LaneNode(path.GetPoint(i), GetNodeRotation(path.cumulativeLengthAtEachVertex[i], _road.EndOfPathInstruction), currRoadNode, prev, null, path.DistanceBetweenPoints(i - 1, i));
```

Comment:
Seems like "distanceToNextNode" is actually being set to the distance to the previous node.